File: week1/code/fibonacci.py
import time

def fib_recur(n):
   if n == 0:
      return 0
   elif n == 1:
      return 1
   else:
      return fib_recur(n - 1) + fib_recur(n - 2)

def fib_iter(n):
   cur_term = 0
   next_term = 1
   total = 0

   for k in range(0, n):
      total = cur_term + next_term
      next_term = cur_term
      cur_term = total

   return total

for i in range(5, 31):
  start_time = time.clock()
  fib_recur(i)
  print("Recur for %d took %f ms" % (i, 1000 * (time.clock() - start_time)))

for i in range(5, 31):
  start_time2 = time.clock()
  fib_iter(i)
  print("Iter for %d took %f ms" % (i, 1000 * (time.clock() - start_time2))) 

File: week1/code/fibonacci.c
#include<stdio.h>

unsigned long long fib_iter(int n);
unsigned long long fib_recur(int n);

int main()
{
  int arr[] = {5, 10, 20, 30, 50};
  
  for (int i = 0; i < 10; i++) {
    //printf("%d ", arr[i]);
    printf("%llu ", fib_iter(arr[i]));
   // printf("%llu ", fib_recur(arr[i]));
  }
  
  return 0;
}

unsigned long long fib_iter(int n) {
  int cur_term = 0;
  int next_term = 1;
  int total = 0;

  for (int i = 0; i <= n; i++) {
    total = cur_term + next_term;
    next_term = cur_term;
    cur_term = total;
  }

  return total;
}

unsigned long long fib_recur(int n) {
  if (n == 0) {
    return 0;
  } else if (n == 1) {
    return 1;
  } else {
    return fib_recur(n - 1) + fib_recur(n - 2);
  }
}
